samples/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgismaps/sample/applymosaicruletorasters/components/ApplyMosaicRuleToRastersViewModel.kt

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applymosaicruletorasters.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.RasterLayer
+import com.arcgismaps.raster.ImageServiceRaster
+import com.arcgismaps.raster.MosaicMethod
+import com.arcgismaps.raster.MosaicOperation
+import com.arcgismaps.raster.MosaicRule
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for the Apply Mosaic Rule to Rasters sample.
+ */
+class ApplyMosaicRuleToRastersViewModel(app: Application) : AndroidViewModel(app) {
+
+    // The map used in the sample, with a topographic basemap.
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISTopographic)
+
+    // The ImageServiceRaster from the raster image service (Amberg, Germany).
+    private val imageServiceRaster = ImageServiceRaster(
+        url = "https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer"
+    )
+
+    // The RasterLayer that displays the ImageServiceRaster on the map.
+    private val rasterLayer = RasterLayer(imageServiceRaster)
+
+    // The list of available mosaic rule types for the dropdown menu.
+    val mosaicRuleTypes = MosaicRuleType.entries
+
+    // The currently selected mosaic rule type.
+    private val _selectedRuleType = MutableStateFlow(MosaicRuleType.ObjectID)
```

Comment:
This does not need to be a flow, We should make this a State instead

    private var _selectedRuleType by mutableStateOf(MosaicRuleType.ObjectID)
    val selectedRuleType: MosaicRuleType
        get() = _selectedRuleType